### PR TITLE
Link Sharing service design

### PR DESCRIPTION
**Pull request by azdagron**
This PR outlines a design for a proposed link sharing service that can be used to share objects in the storage network via a simple URL.

### CODE REVIEW

File: docs/design/link-sharing-service.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+Presumably, with a small change to the uplink library, pieces downloaded from
+storage node operators could presumably be cached locally on disk or via some
+sort of shared cache (in the presence of horizontally scaled link sharing
+services). At this time such caching seems premature and potentially 
+harmful in that:
+
+- It complicates the code.
+- It complicates deployment.
+- It decreases payout potential for storage node operators, reducing incentive.
+- The whitepaper already describes a mechanism to deal with hot objects.
```
Comment (ifraixedes) [inline docs/design/link-sharing-service.md]: I would the section where that's mentioned.

Comment (azdagron) [reply docs/design/link-sharing-service.md]: I've updated the PR with a link to the whitepaper and a mention of the specific section.

File: docs/design/link-sharing-service.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	bytes encrypted_path_prefix = 2;
+}
+
+message Share {
+	string satellite_url = 1;
+
+	string api_key = 2;
+
+	string bucket = 3;
```
Comment (azdagron) [inline docs/design/link-sharing-service.md]: Hmm, looking at other proto's this is defined as `bytes`? What about `api_key`?

Comment (ifraixedes) [reply docs/design/link-sharing-service.md]: If I'm not 100% sure, but the `api_key` is going to be a [macaroon](https://github.com/storj/storj/blob/master/pkg/macaroon/apikey.go#L57) and the [serialize](https://github.com/storj/storj/blob/master/pkg/macaroon/apikey.go#L145) / [deserialize](https://github.com/storj/storj/blob/master/pkg/macaroon/apikey.go#L63) from a string type

**Comment by azdagron** [comment]
Please leave future comments on #2360, which has been updated to accommodate the encryption context and is from a branch on github.com/storj/storj instead of my since-deleted fork.

**Pull request closed without merging**